util/math_reference.cpp

```diff
@@ -295,9 +302,12 @@ int64_t rhadd(const int64_t a, const int64_t b) {
  */
 template <typename T>
 T clamp_t(T v, T minv, T maxv) {
+  if (minv > maxv && std::is_floating_point<T>::value)
+    return NAN;
   return (v < minv) ? minv : ((v > maxv) ? maxv : v);
 }
 
+
```

Comment:
This empty line is strange, is the file clang-formatted?

Reply:
No, it's not
Clang-formatting would affect most of the files or, if run it on only files from this patch, it would affect most of the previous code too. May be it would be better to run clang-formatting and add it with different PR?

Reply:
That makes sense. That's unfortunate, though, I thought the files were clang-formatted.

Reply:
@kopylovanat, you can apply clang-format for patch only. See https://clang.llvm.org/docs/ClangFormat.html#script-for-patch-reformatting.

Another strategy is to create two separate changes:
1. NFC patch with code formatting for existing source file.
2. New functionality already correctly formatted.

@ProGTX, I think it might be useful to integrate code formatting check if we would like CTS to be clang-formatted.

Reply:
Applied clang-format for this patch.